layers.go

```diff
@@ -1554,8 +1554,9 @@ func (r *layerStore) applyDiffWithOptions(to string, layerOptions *LayerOptions,
 	tsdata := bytes.Buffer{}
 	compressor, err := pgzip.NewWriterLevel(&tsdata, pgzip.BestSpeed)
 	if err != nil {
-		compressor = pgzip.NewWriter(&tsdata)
+		return -1, err
 	}
+	defer compressor.Close()
```

Comment:
As @shitou12345678 points out in #1623 , this is incorrect: `.Close` triggers compression of `compressor.currentBuffer`, and writes a trailer. That must happen before consuming `tsdata`.